.golangci.yml

```diff
@@ -17,3 +19,6 @@ issues:
     - linters:
         - staticcheck
       text: "SA9003:"
+    - linters:
+        - revive
+      text: "unused-parameter: parameter '.+' seems to be unused, consider removing or renaming it as _"
```

Comment:
Alternatively we can fix these
```sh
make lint 2>/dev/null | grep unused-parameter | wc -l
33
```

Reply:
For context these are the kind of issues flagged:

```
pkg/decrypter/decrypter_test.go:10:32: unused-parameter: parameter 'secret' seems to be unused, consider removing or renaming it as _ (revive)
func (d mockDecrypter) Decrypt(secret string) (string, error) {
                               ^
pkg/decrypter/decrypter_test.go:16:35: unused-parameter: parameter 'secret' seems to be unused, consider removing or renaming it as _ (revive)
func (d mockErrDecrypter) Decrypt(secret string) (string, error) {
                                  ^
channel/directory.go:27:28: unused-parameter: parameter 'ctx' seems to be unused, consider removing or renaming it as _ (revive)
func (d *Directory) Update(ctx context.Context, logger *log.Entry) error {
                           ^
channel/directory.go:31:29: unused-parameter: parameter 'channel' seems to be unused, consider removing or renaming it as _ (revive)
func (d *Directory) Version(channel string, overrides map[string]string) (ConfigVersion, error) {
                            ^
channel/directory.go:39:25: unused-parameter: parameter 'ctx' seems to be unused, consider removing or renaming it as _ (revive)
func (d *Directory) Get(ctx context.Context, logger *log.Entry) (Config, error) {
                        ^
channel/mock_source.go:23:29: unused-parameter: parameter 'ctx' seems to be unused, consider removing or renaming it as _ (revive)
func (s *mockSource) Update(ctx context.Context, logger *logrus.Entry) error {
                            ^
channel/mock_source.go:27:46: unused-parameter: parameter 'overrides' seems to be unused, consider removing or renaming it as _ (revive)
func (s *mockSource) Version(channel string, overrides map[string]string) (ConfigVersion, error) {
                                             ^
channel/git.go:117:39: unused-parameter: parameter 'overrides' seems to be unused, consider removing or renaming it as _ (revive)
func (g *Git) Version(channel string, overrides map[string]string) (ConfigVersion, error) {
                                      ^
pkg/kubernetes/kubernetes_test.go:63:3: redefines-builtin-id: redefinition of the built-in function delete (revive)
		delete, ok := action.(k8stesting.DeleteAction)
		^
pkg/kubernetes/kubernetes_test.go:72:34: unused-parameter: parameter 'resource' seems to be unused, consider removing or renaming it as _ (revive)
func (f *fakeRESTMapper) KindFor(resource schema.GroupVersionResource) (schema.GroupVersionKind, error) {
                                 ^
pkg/kubernetes/kubernetes_test.go:76:35: unused-parameter: parameter 'resource' seems to be unused, consider removing or renaming it as _ (revive)
func (f *fakeRESTMapper) KindsFor(resource schema.GroupVersionResource) ([]schema.GroupVersionKind, error) {
                                  ^
pkg/kubernetes/kubernetes_test.go:88:39: unused-parameter: parameter 'input' seems to be unused, consider removing or renaming it as _ (revive)
func (f *fakeRESTMapper) ResourcesFor(input schema.GroupVersionResource) ([]schema.GroupVersionResource, error) {
                                      ^
pkg/kubernetes/kubernetes_test.go:92:38: unused-parameter: parameter 'gk' seems to be unused, consider removing or renaming it as _ (revive)
func (f *fakeRESTMapper) RESTMapping(gk schema.GroupKind, versions ...string) (*meta.RESTMapping, error) {
                                     ^
pkg/kubernetes/kubernetes_test.go:96:39: unused-parameter: parameter 'gk' seems to be unused, consider removing or renaming it as _ (revive)
func (f *fakeRESTMapper) RESTMappings(gk schema.GroupKind, versions ...string) ([]*meta.RESTMapping, error) {
                                      ^
pkg/kubernetes/kubernetes_test.go:100:47: unused-parameter: parameter 'resource' seems to be unused, consider removing or renaming it as _ (revive)
func (f *fakeRESTMapper) ResourceSingularizer(resource string) (singular string, err error) {
                                              ^
registry/static.go:12:39: unused-parameter: parameter 'filter' seems to be unused, consider removing or renaming it as _ (revive)
func (r *staticRegistry) ListClusters(filter Filter) ([]*api.Cluster, error) {
                                      ^
registry/static.go:29:40: unused-parameter: parameter 'cluster' seems to be unused, consider removing or renaming it as _ (revive)
func (r *staticRegistry) UpdateCluster(cluster *api.Cluster) error {
                                       ^
registry/file.go:30:37: unused-parameter: parameter 'filter' seems to be unused, consider removing or renaming it as _ (revive)
func (r *fileRegistry) ListClusters(filter Filter) ([]*api.Cluster, error) {
                                    ^
pkg/updatestrategy/rolling_update_test.go:43:55: unused-parameter: parameter 'nodePool' seems to be unused, consider removing or renaming it as _ (revive)
func (m *mockNodePoolManager) MarkPoolForDecommission(nodePool *api.NodePool) error {
                                                      ^
pkg/updatestrategy/clc_update_test.go:36:58: unused-parameter: parameter 'nodePool' seems to be unused, consider removing or renaming it as _ (revive)
func (m *mockNodePoolManagerCLC) MarkPoolForDecommission(nodePool *api.NodePool) error {
                                                         ^
pkg/updatestrategy/node_pool_manager_test.go:44:44: unused-parameter: parameter 'nodePool' seems to be unused, consider removing or renaming it as _ (revive)
func (n *mockProviderNodePoolsBackend) Get(nodePool *api.NodePool) (*NodePool, error) {
                                           ^
pkg/updatestrategy/node_pool_manager_test.go:52:62: unused-parameter: parameter 'decrementDesired' seems to be unused, consider removing or renaming it as _ (revive)
func (n *mockProviderNodePoolsBackend) Terminate(node *Node, decrementDesired bool) error {
                                                             ^
pkg/updatestrategy/aws_asg_test.go:27:53: unused-parameter: parameter 'input' seems to be unused, consider removing or renaming it as _ (revive)
func (a *mockASGAPI) DescribeAutoScalingGroupsPages(input *autoscaling.DescribeAutoScalingGroupsInput, fn func(*autoscaling.DescribeAutoScalingGroupsOutput, bool) bool) error {
                                                    ^
pkg/updatestrategy/aws_asg_test.go:32:48: unused-parameter: parameter 'input' seems to be unused, consider removing or renaming it as _ (revive)
func (a *mockASGAPI) DescribeAutoScalingGroups(input *autoscaling.DescribeAutoScalingGroupsInput) (*autoscaling.DescribeAutoScalingGroupsOutput, error) {
                                               ^
controller/controller_test.go:43:39: unused-parameter: parameter 'cluster' seems to be unused, consider removing or renaming it as _ (revive)
func (p *mockErrProvisioner) Supports(cluster *api.Cluster) bool {
                                      ^
controller/controller_test.go:57:45: unused-parameter: parameter 'cluster' seems to be unused, consider removing or renaming it as _ (revive)
func (p *mockErrCreateProvisioner) Supports(cluster *api.Cluster) bool {
                                            ^
controller/controller_test.go:85:37: unused-parameter: parameter 'filter' seems to be unused, consider removing or renaming it as _ (revive)
func (r *mockRegistry) ListClusters(filter registry.Filter) ([]*api.Cluster, error) {
                                    ^
controller/controller_test.go:120:36: unused-parameter: parameter 'logger' seems to be unused, consider removing or renaming it as _ (revive)
func (r *mockChannelSource) Delete(logger *log.Entry, config channel.Config) error {
                                   ^
controller/controller_test.go:143:36: unused-parameter: parameter 'manifestName' seems to be unused, consider removing or renaming it as _ (revive)
func (c *mockConfig) StackManifest(manifestName string) (channel.Manifest, error) {
                                   ^
controller/controller_test.go:147:35: unused-parameter: parameter 'manifestName' seems to be unused, consider removing or renaming it as _ (revive)
func (c *mockConfig) EtcdManifest(manifestName string) (channel.Manifest, error) {
                                  ^
controller/controller_test.go:151:39: unused-parameter: parameter 'profileName' seems to be unused, consider removing or renaming it as _ (revive)
func (c *mockConfig) NodePoolManifest(profileName string, manifestName string) (channel.Manifest, error) {
                                      ^
provisioner/clusterpy_test.go:281:36: unused-parameter: parameter 'manifestName' seems to be unused, consider removing or renaming it as _ (revive)
func (c *mockConfig) StackManifest(manifestName string) (channel.Manifest, error) {
                                   ^
provisioner/clusterpy_test.go:289:39: unused-parameter: parameter 'profileName' seems to be unused, consider removing or renaming it as _ (revive)
func (c *mockConfig) NodePoolManifest(profileName string, manifestName string) (channel.Manifest, error) {
                                      ^
provisioner/template_test.go:15:13: unused-parameter: parameter 't' seems to be unused, consider removing or renaming it as _ (revive)
func render(t *testing.T, templates map[string]string, templateName string, data interface{}, adapter *awsAdapter) (string, error) {
```

I would propose we fix them

Reply:
Looks like there are many more once you start fixing them so I guess it is fine to exclude this rule

Reply:
Renamed in the second commit so we can choose the way